File: 4.median-of-two-sorted-arrays.cpp
/*
 * @lc app=leetcode id=4 lang=cpp
 *
 * [4] Median of Two Sorted Arrays
 */

// @lc code=start
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int n = nums1.size();
        int m = nums2.size();
        int x = 0; 
        int y = 0;
        vector<int> a;
        while(x<n&&y<m){
            if(nums1[x]<=nums2[y]){
                a.push_back(nums1[x]);
                x++;
            }
            else{
                a.push_back(nums2[y]);
                y++;
            }
        }
        while(x<n){
            a.push_back(nums1[x]);
            x++;
        }
        while(y<m){
            a.push_back(nums2[y]);
            y++;
        }
        if((n+m)%2==0){
            if((n+m)==2){
                return float((float(a[0])+float(a[1]))/float(2));
            }
            else{
                int z = (n+m)/2; 
                return float((float(a[z])+float(a[z-1]))/float(2));
            }
            
        }
        else{
            int z = (n+m)/2;
            return float(a[z]);
        }
        return 0.0;
    }
};
// @lc code=end

File: 4.median-of-two-sorted-arrays.cs
/*
 * @lc app=leetcode id=4 lang=csharp
 *
 * [4] Median of Two Sorted Arrays
 */

// @lc code=start
public class Solution {
    public double FindMedianSortedArrays(int[] nums1, int[] nums2) {
        List<int> a = new List<int>();
        int n = nums1.Length;
        int m = nums2.Length;
        int i=0, j=0;
        while(i<n&&j<m){
            if(nums1[i]>nums2[j]){
                a.Add(nums2[j]);
                j++;
            }
            else{
                a.Add(nums1[i]);
                i++;
            }
        }
        while(i<n){
            a.Add(nums1[i]);
            i++;
        }
        while(j<m){
            a.Add(nums2[j]);
            j++;
        }
        if((n+m)%2==0){
            int y = (n+m)/2;
            return ((double)a[y]+(double)a[y-1])/2.0;
        }
        else{
            int y = (n+m)/2;
            return (double)a[y];
        }
        return 0.0;
    }
}
// @lc code=end

